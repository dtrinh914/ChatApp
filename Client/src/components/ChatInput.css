.ChatInput{
    height: 10vh;
    display: flex;
}
.ChatInput textarea{
    flex-grow: 4;
    resize: none;
}
.ChatInput button{
    flex-grow: 1;
}