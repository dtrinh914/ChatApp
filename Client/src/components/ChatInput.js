import React from 'react';
import useInput from '../hooks/useInput';
import './ChatInput.css'

function ChatInput(props){
    const [message, setMessage, clearMessage] = useInput();

    function handleClick(){
        props.newMessage(message);
    }

    function handleSumbit(e){
        e.preventDefault()
        clearMessage();
    }
    return(
        <form className='ChatInput' onSubmit={handleSumbit}>
            <input type="text" value={message} onChange={setMessage} />
            <button onClick={handleClick}>Send</button>
        </form>
    );
}

export default ChatInput;