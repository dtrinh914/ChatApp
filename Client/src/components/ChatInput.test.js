import React from 'react';
import { render } from '@testing-library/react';
import ChatInput from './ChatInput';

test('renders ChatInput', () => {
  const input = render(<ChatInput />);
  expect(input.toJSON()).toMatchSnapsot();
});
