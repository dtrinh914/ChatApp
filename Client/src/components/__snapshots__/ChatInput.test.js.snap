// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="ChatInput"
  >
    <input
      type="text"
      value=""
    />
    <button>
      Send
    </button>
  </form>
</DocumentFragment>
`;
